### ISSUE TITLE
Off-by-one in VPA CSI sequence

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dragonlinux/connectbot#582]
```
What steps will reproduce the problem?

 $ echo -e "\e[3dTHIS IS THE THIRD LINE"

Expected: THIS IS THE THIRD LINE   printed on the 3rd line of the screen
Actual:   THIS IS THE THIRD LINE   printed on the 4th line of the screen

Similarly, "\e[1d" should go to the top line of the display but it goes to the 
second, requiring "\e[0d" to actually go to the topmost.


What version of the product are you using (you can see this by using Menu
-> About in the Host List)?

ConnectBot 1.7.1 (v1.7.1 2010.10.08)


What type of system are you trying to connect to?

Linux, debian.

If you are able to connect, what is the output of "echo $TERM", "uname -a",
and any other relevant information on the host?

 $ echo $TERM
 xterm

```

Original issue reported on code.google.com by `pev...@google.com` on 8 Jun 2012 at 11:46

**Comment by GoogleCodeExporter** [comment]
```
http://code.google.com/p/connectbot/source/browse/src/de/mud/terminal/vt320.java
#2593

I suspect this line ought to read

  R = DCEvars[0] - 1;
```

Original comment by `pev...@google.com` on 8 Jun 2012 at 12:08

**Comment by GoogleCodeExporter** [comment]
Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 1:33
- Added labels: **Component-Terminal**

**Comment by GoogleCodeExporter** [comment]
Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 1:34
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 53023daeee79.
```

Original comment by `leon...@leonerd.org.uk` on 29 Sep 2012 at 1:53
- Changed state: **Fixed**
